Umbraco-Heartcore/Tutorials/Creating-A-Custom-Grid-Editor/index.md

```diff
@@ -0,0 +1,454 @@
+
+We now have the **Add image** button rendering, but clicking it does nothing. Let's do something about that. 
+
+* Go back to the **Settings** section.
+* Click on the **Image Gallery** Grid Editor.
+
+We will start by adding the following to the top of the file:
+
+```javascript
+import { mediaPicker } from 'https://cdn.jsdelivr.net/npm/@umbraco/headless-backoffice-bridge@0/headless-backoffice-bridge.min.js'
+```
+
+This imports the `mediaPicker` function from the [backoffice bridge](https://github.com/umbraco/Umbraco.Headless.Backoffice.Bridge) library.
+
+We will add another function before the `render` function:
+
+```javascript
+  showPicker() {
+    mediaPicker.show({
+      disableFolderSelect: true,
+      onlyImages: true,
+      showDetails: true,
+      submit: (items) => {
+        const selected = items[0]
+        if(!this.value) this.value = []
+        this.value.push(({ url: selected.udi, ...selected }))
+        this.requestUpdate('value')
+      }
+    })
+  }
+```
+
+This will open the Media Picker with a configuration that only allows selecting images and showing the detail view, which allows us to enter an alt text, caption and choose a focal point.
+
+The submit callback is called when the **Submit** is clicked. It will contain an array of the selected items. In this case it will contain a single item, but if we set `multiple` to `true` multiple items can be returned.
+
+In our callback method we check if `this.value` has been initialized. If it has not we initialize a new array. We then push the selected image to the array. Note that we also store the `udi` (the id of the media) in a `url` property. We will come back to why we do that later.
+
+Since we add to the array and do not set the `value` property we need to tell Lit that the property was updated. We do this by calling `this.requestUpdate('value')`.
+
+We will also need to add a `click` event to the button so it will show the dialog:
+
+```html
+<button type="button" class="btn-reset placeholder" @click=${() => this.showPicker()}>
+```
+
+::note
+Clicking the button in preview mode does nothing. This is because of a limitation when previewing where we do not have access to the full backoffice. Saving and testing on a Content item should work.
+::
+
+* Click **Save**.
+* Go back to the content page.
+* Click the `Add image` button.
+
+We should now see the image overlay.
+
+![Select media overlay shown when clicking the Add image button](images/select-media-overlay.png)
+
+Selecting an image does not currently do anything. Let's do something about that. Lets go back to the grid editor code in the **Settings** section.
+
+Go back to the **Content** section try to add some images to a page using the editor.
+
+We now have an editor that we can add images to which gets rendered in the backoffice. Next step would be to add edit and delete functionality, but we will leave that to the reader.
```

Comment:
Could it be that some steps are missing here? 🤔 

"Let's go back to the Grid Editor code in the **Settings** section"
And then immediately after: "Go back to the **Content** section ... "

Do some steps need to be taken in the Settings section here before going back to the Content section?
Or perhaps it is not even necessary to go to the Settings section 🤔 

I need some help figuring this one out! 😄

Reply:
Maybe just drop the "back" part, so its more about going to the specific section instead of returning to somewhere?